### PR TITLE
[IMP] web: update owl from 2.0.9 to 2.1.1

### PR DESCRIPTION
**Pull request by ged-odoo**
This update contains the following commits:

[FIX] components: solve missing update (concurrency issue) [DOC] remove old references to comp for useRef hook [FIX] devtools: Fix build commands for windows users [FIX] devtools: Fix devtools in detached window
[IMP] devtools: Add a button to navigate to the doc [IMP] devtools: Add devtools documentation
[IMP] Allow app to be mounted in shadow DOM
[IMP] playground: allow sharing playground links
[IMP] runtime: allow validating object values using a type description [FIX] devtools: Increase vertical padding of the search bar [FIX] devtools: Hide the collapse all button
[FIX] devtools: Fix bad computation of highlight boxes on the page [FIX] package: Auto-update package-lock.json

Notes: https://github.com/odoo/owl/releases/tag/v2.1.1

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by ged-odoo** [comment]
@robodoo override=ci/security
Reason: the real diff is actually quite small, but one of the commit (https://github.com/odoo/owl/commit/c0c40c038740c9cc79af9223ead5242ba4f2dbfb) changed some import, so the dependency graph was slightly changed. Because of that, rollup (the tool that package the js files into one single file) has to slightly reorder the inlined content files to satisfy the new dependency ordering. And because of that, the line that defined the markup function was moved, which triggered this ci check.

So, in short: this is just a move
